pkg/app/app.go

```diff
@@ -77,7 +78,12 @@ func defaultFuncOrganization(engine workflow.Engine, config configuration.Config
 	return callback
 }
 
-func defaultFuncApiUrl(config configuration.Configuration, logger *zerolog.Logger) configuration.DefaultValueFunction {
+func defaultFuncApiUrl(engine workflow.Engine, config configuration.Configuration, logger *zerolog.Logger) configuration.DefaultValueFunction {
+	// TODO: implement proper caching
+	// cache check needed in case an invalid PAT is already configured in env var or config file
+	is_pat_valid_key := fmt.Sprintf("%s_%s", auth.CACHED_PAT_IS_VALID_KEY_PREFIX, config.GetString(configuration.AUTHENTICATION_TOKEN))
+	// initial state assumes it's valid
+	config.Set(is_pat_valid_key, true)
```

Comment:
Suggestion: after having the config caching in GAF, let's remove this caching workaround.